### PR TITLE
fix(Script/Instance): Ulduar - Yogg-Saron hardcode & some adustments.

### LINKED ISSUE #11859: Move hardcoded texts of boss_yoggsaron (Ulduar)to the Database
### Current Behaviour

[boss_yoggsaron.cpp](https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp)

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L487

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L587

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L592

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L604

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L609

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L1082

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L1138

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L1184

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L1981

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L1985

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L1989

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L1993

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L1997

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L2001

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L2095

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L2103

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L2107

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L2115

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L2119

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L2123

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L2213

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L2218

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L2222

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L2227

https://github.com/azerothcore/azerothcore-wotlk/blob/b801274b895bbafc222f18a38d946dbf2b307b58/src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp#L2231

### Expected Blizzlike Behaviour

none

### Source

_No response_

### Steps to reproduce the problem

none

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/azerothcore/azerothcore-wotlk/commit/305a2689e965346dc7c12a87650fab08696b0a10

### Operating system

Ubuntu20.04

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by avarishd**
Fixes #11859

## Changes Proposed:
- Added missing creature_text to Yogg.
- Fixed Sara going under the map
- Fixed Death Orbs (Death Ray)
- Added missing RP and texts.

## Issues Addressed:

- Closes #11859

## Tests Performed:

- tested in-game

## How to Test the Changes:

Test with 2+ characters
1. delete cache!
2. Comment out (So you can straight pull Yogg)
` if (m_pInstance->GetData(TYPE_VEZAX) != DONE || m_pInstance->GetData(TYPE_XT002) != DONE)
                return;`
3. .aura 64169 - Sanity aura
4. .go xyz 1866 -23 335 603 - tp in front of Yogg
Check the gossip, check Sara position in Phase 2, check Death Ray's movement

## Known Issues and TODO List:

- [x] fix znCN locale (allegedly)
- [x] Fix death ray being idle

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Maelthyrr** [comment]
Tested and everything looks fine. Thanks for your PR ;)

**Review by UltraNix — APPROVED** [review]

File: src/server/game/Spells/SpellInfoCorrections.cpp [hunk 1]
```diff
@@ -4471,6 +4471,12 @@ void SpellMgr::LoadSpellInfoCorrections()
         spellInfo->AttributesEx2 |= SPELL_ATTR2_IGNORE_LINE_OF_SIGHT;
     });
 
+    // Yogg-Saron - Death Orb - Death Ray Warning Visual & Death Ray Damage Visual
```
Comment (Kitzunu) [inline src/server/game/Spells/SpellInfoCorrections.cpp:4474]:
```suggestion
    // Death Ray Warning Visual, Death Ray Damage Visual
```

Comment (Kitzunu) [reply src/server/game/Spells/SpellInfoCorrections.cpp:4474]: We only use Spell Name in this file

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: src/server/game/Spells/SpellInfoCorrections.cpp (hunk 1 above)
Comment (avarishd) [reply src/server/game/Spells/SpellInfoCorrections.cpp:4474]: Good to know!

**Review by Kitzunu — APPROVED** [review]

**Comment by pklloveyou** [comment]
Tested

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp [hunk 2]
```diff
@@ -429,6 +445,18 @@ class boss_yoggsaron_sara : public CreatureScript
 
         void Reset() override
         {
+            if (!_secondPhase) // Phase 1 wipe
+            {
+                Map::PlayerList const& pList = me->GetMap()->GetPlayers();
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp:450]: you can use instance->DoForAllPlayers()

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp [hunk 3]
```diff
@@ -577,38 +605,12 @@ class boss_yoggsaron_sara : public CreatureScript
...
-            {
-                me->Yell("Could they have been saved?", LANG_UNIVERSAL);
-                me->PlayDirectSound(SARA_P1_KILL2);
-            }
-            else
-            {
-                me->Yell("Powerless to act...", LANG_UNIVERSAL);
-                me->PlayDirectSound(SARA_P1_KILL1);
-            }
+            if (who->GetTypeId() == TYPEID_PLAYER)
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp:610]: who->IsPlayer()

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp (hunk 2 above)
Comment (avarishd) [reply src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp]: I like that one, didn't even know it existed.

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp [hunk 4]
```diff
@@ -429,6 +444,17 @@ class boss_yoggsaron_sara : public CreatureScript
 
         void Reset() override
         {
+            if (!_secondPhase) // Phase 1 wipe
+            {
+                me->GetMap()->DoForAllPlayers([&](Player* player)
+                {
+                    if(Creature* voice = me->FindNearestCreature(NPC_VOICE_OF_YOGG_SARON, 10))
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp:451]:
```suggestion
                    if (Creature* voice = me->FindNearestCreature(NPC_VOICE_OF_YOGG_SARON, 10.0f))
```

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp [hunk 5]
```diff
@@ -577,38 +602,12 @@ class boss_yoggsaron_sara : public CreatureScript
...
-            }
-            else
-            {
-                me->Yell("Powerless to act...", LANG_UNIVERSAL);
-                me->PlayDirectSound(SARA_P1_KILL1);
-            }
+            if (who->IsPlayer())
+                {
+                    Talk(SAY_SARA_KILL);
+                }
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_yoggsaron.cpp:610]:
```suggestion
            {
                Talk(SAY_SARA_KILL);
            }
```

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**